frontend/src/components/RecommendationList.jsx

```diff
@@ -2,53 +2,89 @@ import React, { useState, useEffect } from "react";
 import { Link, useLocation } from "react-router-dom";
 import axios from "axios";
 
-// comment out the 2 console logs to get a glimpse of what's in there :)
-
 const RecommendationList = () => {
   const [myData, setMyData] = useState(null);
+  const [filters, setFilters] = useState({});
   const navigationLocation = useLocation();
-  // console.log(navigationLocation);
 
   useEffect(() => {
-    // forward the values of the form submission and send a get request with them
     // localhost will be changed, just here while in development
     const getData = async () => {
+      // take out the query string from the URL and create an object out of those values
+      const queryString = new URLSearchParams(navigationLocation.search);
+      const queryObject = {};
+      queryString.forEach(function (value, key) {
+        queryObject[key] = value;
+      });
+
+      // --ask backend if it's ok with the frontend passing every value as a string
+      // queryObject.price_eur = queryObject.price_eur.split(",");
+      // queryObject.wine_type = queryObject.wine_type.split(",");
+      // queryObject.origin = queryObject.origin.split(",");
+      // queryObject.flavor_profile = queryObject.flavor_profile.split(",");
+
       const response = await axios.get("http://localhost:8080/api/wines", {
-        params: navigationLocation.state,
+        params: queryObject,
       });
       setMyData(response.data);
-      // console.log(response.data);
+      setFilters(queryObject);
     };
     getData();
-  }, [navigationLocation.state]);
+  }, [navigationLocation.search]);
 
   return (
     <main>
-      <h1>We recommend you</h1>
-      <p>
-        <span>a dry,</span>
-        <span>white wine</span>
-        <span>from France</span>
-      </p>
-
-      <section>
-        <h2>such as</h2>
-        <div>
+      <figure>
+        <p>Your search filters:</p>
+
+        {filters.food_name && <p>{filters.food_name}</p>}
+        {filters.price_eur && filters.price_eur.includes("low") && <p>€</p>}
```

Comment:
I'll take this snippet of code to talk about something important about software development, that affects maintainability and possible bugs in the future. (also what I'll explain I'd recommend to do in any job interview and explain why)
When we strings, name or any type of value that is used across different files, this could also apply to the same file for sure but it's specially important different files, we should organize the code to have a single source of truth to avoid any bugs.
To do that we could have a separate file to group many values like `constants.js` or have the values extracted in the same file where it's being used and makes more sense to be grouped together.
When you extract these constants, you could have something like:
```js
// it's common to have constant as capital case, but it's not mandatory
// you could name this wineQueryParams
const WINE_QUERY_PARAMS = {
  foodName: 'food_name', // string
  priceEur: 'price_eur', // array of PRICE_EUR
  vegan: 'vegan', // boolean
  wineType: 'wine_type', // string
  countryName: 'country_name', // string
  flavorProfile: 'flavor_profile', // array of FLAVOR_PROFILE
};

const PRICE_EUR = {
  low: 'low,
  // the name of the value doesn't need to match exactly,
  // we can represent the value with more meaningful name
  medium: 'med',
  heigh: 'high',
  expensive: 'exp',
};

const FLAVOR_PROFILE = ...
```
This grouping of the name values is what we call [Enum](https://en.wikipedia.org/wiki/Enumerated_type).
And what we  get for writing more code to organize it like that?
Imagine that in the future another developer (or even you, because we as developers will eventually forget about minor details about the code) has to change something in the form in `frontend/src/components/RecommendationForm.jsx`.
It could the name of an input, or it could the value of one of the prices, from `exp` to `expensive` because the because the backend changed the value of the filter, or add a new value to the `flavor_profile`, and this developer has no idea that all these things are tightly coupled to the `frontend/src/components/RecommendationList.jsx`.
A bug could be easily introduced in this scenario. And it's not the fault of the developer that changed it, because they didn't have any idea that the values were connected outside the file.
We can be more future proof by having a single source of truth for things that are connected.
That's why I freak out when I see raw strings being used when they represent specific and defined things in the system. So much can go wrong.